cranelift-wasm/src/code_translator.rs

```diff
@@ -270,9 +271,14 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
 
             if !builder.is_unreachable() || !builder.is_pristine() {
                 let return_count = frame.num_return_values();
-                builder
-                    .ins()
-                    .jump(frame.following_code(), state.peekn(return_count));
+                let args = bitcast_arguments(
+                    state.peekn(return_count),
+                    &builder.func.signature.returns.clone(), // TODO this use of signature
+                    // returns may be incorrect; what if this End is in a different control flow
+                    // structure?
```

Comment:
This PR is still a WIP because 1) I'm not yet confident that I've covered all of the possible cases where a vector should be bitcast to another type and 2) in situations like the above (`Operator::End`), I am pretending that I can bitcast to the function return types but in fact the `End` may be the end of some other block, not necessarily the end of a function (though adding this makes more SIMD spec tests pass).

Reply:
I think this might be on the right track, except that you should use the block's signature, rather than the function's signature.